library/src/main/java/com/google/android/exoplayer/extractor/flv/FlvExtractor.java

```diff
@@ -0,0 +1,310 @@
+      long timestamp = tagHeaderBuffer.readUnsignedInt24();
+      timestamp = (tagHeaderBuffer.readUnsignedByte() << 24) | timestamp;
+      int streamId = tagHeaderBuffer.readUnsignedInt24();
+
+      currentTagHeader.type = type;
+      currentTagHeader.dataSize = dataSize;
+      currentTagHeader.timestamp = timestamp * 1000;
+      currentTagHeader.streamId = streamId;
+
+      // Sanity checks.
+      Assertions.checkState(type == TAG_TYPE_AUDIO || type == TAG_TYPE_VIDEO
+          || type == TAG_TYPE_SCRIPT_DATA);
+      Assertions.checkState(dataSize <= Integer.MAX_VALUE);
+      // Reuse tagData buffer to avoid lot of memory allocation (performance penalty).
+      if (tagData == null || dataSize > tagData.capacity()) {
+        tagData = new ParsableByteArray((int) dataSize);
+      } else {
+        tagData.setPosition(0);
+      }
+      tagData.setLimit(dataSize);
+      parserState = STATE_READING_SAMPLE;
+
+    } catch (EOFException eof) {
+      return false;
+    }
+
+    return true;
+  }
+
+  /**
+   * Reads payload of an FLV tag from the provided {@link ExtractorInput}.
+   * @param input The {@link ExtractorInput} from which to read.
+   * @param seekPosition
+   * @return One of {@link Extractor#RESULT_CONTINUE} and {@link Extractor#RESULT_END_OF_INPUT}.
+   * @throws IOException If an error occurred reading from the source.
+   * @throws InterruptedException If the thread was interrupted.
+   * @throws TagPayloadReader.UnsupportedTrack If payload of the tag is using a codec non
+   * supported codec.
+   */
+  private int readSample(ExtractorInput input, PositionHolder seekPosition) throws IOException,
+      InterruptedException, AudioTagPayloadReader.UnsupportedTrack {
+    if (tagData != null) {
+      if (!input.readFully(tagData.data, 0, currentTagHeader.dataSize, true)) {
+        return RESULT_END_OF_INPUT;
+      }
+      tagData.setPosition(0);
+    } else {
+      input.skipFully(currentTagHeader.dataSize);
+      return RESULT_CONTINUE;
+    }
+
+    // Pass payload to the right payload reader.
+    if (currentTagHeader.type == TAG_TYPE_AUDIO && audioReader != null) {
+      audioReader.consume(tagData, currentTagHeader.timestamp);
+    } else if (currentTagHeader.type == TAG_TYPE_VIDEO && videoReader != null) {
+      videoReader.consume(tagData, currentTagHeader.timestamp);
+    } else if (currentTagHeader.type == TAG_TYPE_SCRIPT_DATA && metadataReader != null) {
+      metadataReader.consume(tagData, currentTagHeader.timestamp);
+      if (metadataReader.durationUs != C.UNKNOWN_TIME_US) {
+        if (audioReader != null) {
+          audioReader.durationUs = metadataReader.durationUs;
+        }
+        if (videoReader != null) {
+          videoReader.durationUs = metadataReader.durationUs;
+        }
+      }
+    } else {
+      tagData.reset();
+    }
+
+    parserState = STATE_READING_TAG_HEADER;
+
+    return RESULT_CONTINUE;
+  }
+
+  // SeekMap implementation.
+  // TODO: Add seeking support
```

Comment:
In fact, when playing an FLV file in a browser (and then, using a Flash based player) you will notice that player is only able to seek to the portion of the file that was already downloaded. There is no way of seeking to a position that was not previously downloaded as there is no way of calculating bytes given a timestamp. 

As a solution to that, most of CDN's provide some kind of "virtual seeking" (probably not the best name). This is stream url accepting an optional query string parameter with the seeking offset (Ex: http://domain/file.flv?time_offset=xxxx) and then, CDN analyzing the file and returning FLV tags from the position specified in the query string parameter.

Assuming constant bitrate could be enough when working with only audio (I implemented something like this in the CDN side when working with MP3 CBR streams without access to MP3 "seeking headers") but it accuracy would be really poor when working with video tracks. With H.264 baseline profile you can do something, but not accurate at all, if you even use more complicated stuffs like main profile and bidirectional frames, any assumption you can take regarding time <-> byte will be wrong. 

Anyway, in the specific case of FLV, there is not a sync sequence as you can find in other formats, so there is not a very good way of re-sync once you jump into the middle of the stream.